actions/admin/transactions.go

```diff
@@ -4,81 +4,84 @@ import (
 	"net/http"
 
 	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/internal/query"
 	"github.com/bitcoin-sv/spv-wallet/mappings"
-	"github.com/bitcoin-sv/spv-wallet/models"
 	"github.com/bitcoin-sv/spv-wallet/models/filter"
+	"github.com/bitcoin-sv/spv-wallet/models/response"
 	"github.com/bitcoin-sv/spv-wallet/server/reqctx"
 	"github.com/gin-gonic/gin"
 )
 
-// transactionsSearch will fetch a list of transactions filtered by metadata
-// Search for transactions filtering by metadata godoc
-// @Summary		Search for transactions
-// @Description	Search for transactions
+// adminGetTxByID fetches a transaction by id for admins
+// @Summary		Get transaction by id for admins
+// @Description	Get transaction by id for admins
 // @Tags		Admin
 // @Produce		json
-// @Param		SearchTransactions body filter.SearchTransactions false "Supports targeted resource searches with filters and metadata, plus options for pagination and sorting to streamline data exploration and analysis"
-// @Success		200 {object} []models.Transaction "List of transactions"
-// @Failure		400	"Bad request - Error while parsing SearchTransactions from request body"
-// @Failure 	500	"Internal server error - Error while searching for transactions"
-// @Router		/v1/admin/transactions/search [post]
+// @Param		id path string true "Transaction ID"
+// @Success		200 {object} response.Transaction "Transaction"
+// @Failure		400	"Bad request - Transaction not found or error in data fetching"
+// @Failure 	500	"Internal Server Error - Error while fetching transaction"
+// @Router		/api/v1/admin/transactions/{id} [get]
 // @Security	x-auth-xpub
-func transactionsSearch(c *gin.Context, _ *reqctx.AdminContext) {
+func adminGetTxByID(c *gin.Context, _ *reqctx.AdminContext) {
 	logger := reqctx.Logger(c)
-	var reqParams filter.SearchTransactions
-	if err := c.Bind(&reqParams); err != nil {
-		spverrors.ErrorResponse(c, spverrors.ErrCannotBindRequest, logger)
-		return
-	}
+	id := c.Param("id")
 
-	transactions, err := reqctx.Engine(c).GetTransactions(
-		c.Request.Context(),
-		mappings.MapToMetadata(reqParams.Metadata),
-		reqParams.Conditions.ToDbConditions(),
-		mappings.MapToQueryParams(reqParams.QueryParams),
-	)
+	transaction, err := reqctx.Engine(c).GetAdminTransaction(c, id)
 	if err != nil {
-		spverrors.ErrorResponse(c, err, logger)
+		spverrors.ErrorResponse(c, spverrors.ErrCouldNotFindTransaction.WithTrace(err), logger)
 		return
 	}
-
-	contracts := make([]*models.Transaction, 0)
-	for _, transaction := range transactions {
-		contracts = append(contracts, mappings.MapToOldTransactionContractForAdmin(transaction))
+	if transaction == nil {
+		spverrors.ErrorResponse(c, spverrors.ErrCouldNotFindTransaction, logger)
```

Comment:
Please also wrap it here

Reply:
What to Wrap? In this case there is no `error` - just `transaction` is nil which indicates that "Could not find trasnaction" - thats - all.
Looks Good To Me 🙈

Reply:
Right, misread the line above 🙈